packages/react-next/src/components/Image/Image.base.tsx

```diff
@@ -0,0 +1,173 @@
+import { useMergedRefs } from '@uifabric/react-hooks';
+
+const getClassNames = classNamesFunction<IImageStyleProps, IImageStyles>();
+
+export interface IImageState {
+  loadState?: ImageLoadState;
+}
+
+const SVG_REGEX = /\.svg$/i;
+const KEY_PREFIX = 'fabricImage';
+
+function useLoadState(
+  props: IImageProps,
+  imageElement: React.RefObject<HTMLImageElement>,
+  frameElement: React.RefObject<HTMLDivElement>,
+) {
+  const [loadState, setLoadState] = React.useState<ImageLoadState>(ImageLoadState.notLoaded);
+
+  React.useLayoutEffect(() => {
+    // If the src property changes, reset the load state
+    if (loadState !== ImageLoadState.notLoaded) {
+      setLoadState(ImageLoadState.notLoaded);
+    }
+  }, [props.src]);
+
+  React.useEffect(() => {
+    if (loadState === ImageLoadState.notLoaded) {
+      // testing if naturalWidth and naturalHeight are greater than zero is better than checking
+      // .complete, because .complete will also be set to true if the image breaks. However,
+      // for some browsers, SVG images do not have a naturalWidth or naturalHeight, so fall back
+      // to checking .complete for these images.
+      const isLoaded: boolean = imageElement.current
+        ? (props.src && imageElement.current.naturalWidth > 0 && imageElement.current.naturalHeight > 0) ||
+          (imageElement.current.complete && SVG_REGEX.test(props.src!))
+        : false;
+
+      if (isLoaded) {
+        setLoadState(ImageLoadState.loaded);
+      }
+    }
+  });
+
+  React.useEffect(() => {
+    props.onLoadingStateChange?.(loadState);
+  }, [loadState]);
+
+  const onImageLoaded = React.useCallback(
+    (ev: React.SyntheticEvent<HTMLImageElement>) => {
+      props.onLoad?.(ev);
+      if (props.src) {
+        setLoadState(ImageLoadState.loaded);
+      }
+    },
+    [props.src, props.onLoad],
+  );
+
+  const onImageError = React.useCallback(
+    (ev: React.SyntheticEvent<HTMLImageElement>) => {
+      props.onError?.(ev);
+      setLoadState(ImageLoadState.error);
+    },
+    [props.src, props.onError],
+  );
+
+  return [loadState, onImageLoaded, onImageError] as const;
+}
+
+export const ImageBase = React.forwardRef((props: IImageProps, forwardedRef: React.Ref<HTMLImageElement>) => {
+  const frameElement = React.useRef<HTMLDivElement>() as React.RefObject<HTMLDivElement>;
+  const imageElement = React.useRef<HTMLImageElement>() as React.RefObject<HTMLImageElement>;
+  const [loadState, onImageLoaded, onImageError] = useLoadState(props, imageElement, frameElement);
+
+  const imageProps = getNativeProps<React.ImgHTMLAttributes<HTMLImageElement>>(props, imgProperties, [
+    'width',
+    'height',
+  ]);
+  const {
+    src,
+    alt,
+    width,
+    height,
+    shouldFadeIn = true,
+    shouldStartVisible,
+    className,
+    imageFit,
+    role,
+    maximizeFrame,
+    styles,
+    theme,
+  } = props;
+  const coverStyle = computeCoverStyle(props, loadState, imageElement, frameElement);
```

Comment:
this is an expensive call. i dont think we should call only if previous state is not loaded and current state is loaded (looks like that's the logic in original code too)

I wrote a hook a while back for getting previous state:
```
import * as React from "react";

// Reference: https://reactjs.org/docs/hooks-faq.html#how-to-get-the-previous-props-or-state
// It's possible that in the future React will provide a usePrevious Hook out of the box since it's a relatively common use case.

export function usePrevious<T>(value: T | undefined): T | undefined {
  // The ref object is a generic container whose current property is mutable ...
  // ... and can hold any value, similar to an instance property on a class
  const ref = React.useRef<T | undefined>(undefined);

  // Store current value in ref
  React.useEffect(() => {
    ref.current = value;
  }, [value]); // Only re-run if value changes

  // Return previous value (happens before update in useEffect above)
  return ref.current;
}
```